### PR TITLE
CVPR SAIAD Submission

### PR DESCRIPTION
**Pull request by axkoenig**
Here is the code for our CVPR SAIAD Workshop submission. I reviewed the code one last time, brushing up last bits and pieces. This PR will also serve as a means for people to check the code diff between our work and the original repository.

The `README.MD` should contain all the relevant information for reproducing the experiments and gives an overview of the repo. It would be a good start to review the `README.MD`, and move on to the code from there.

Cheers, Alex

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,186 +1,149 @@
...
- from `STEGO\src` to generate the prerequisite KNN information for the custom dataset.
- 
-You can now train on your custom dataset using:
-```shell script
-python train_segmentation.py
+Now we train the STEGO segmentation head and the other dimensionality reduction techniques for each dataset. For STEGO, we simply execute the provided `train_segmentation.py` script with different target dimensions to ensure maximum reproducibility.
+
+```bash
+# COCOSTUFF
+for dim in 90 768 384 192 100 96 48 24 12 6 3:
```
Comment (MaxSchambach) [inline README.md:83]: After a bash for loop, there is a `;`, not a `:`, please update all of them :)

File: README.md [hunk 2]
```diff
@@ -1,186 +1,149 @@
...
-	url={https://openreview.net/forum?id=SaKO6z6Hl0c}
-}
+
+We wrote a variant `mxm_train_segmentation.py` of the original `train_segmentation.py` script that uses the pre-computed DINO features instead of running the usual DINO forward pass. It was easier to copy the original `train_segmentation.py` and create a new `mxm_train_segmentation.py` with these more invasive additions instead of editing the original file and keeping it backward compatible. 
+
+Now you can run the remaining benchmarks on the other dimension reduction baselines. Once you start the script, the dimension reduction techniques (i.e., PCA, RP) will be automatically fitted on a subset of the training dataset. For HNNE, we need not only fit the method but also pre-compute its down-projected features because calculating them on the fly in the model's forward pass is prohibitively expensive. Hence, to reproduce the HNNE results from the appendix, run `python mxm_dimred.py dataset_name="cityscapes"`. Consequently, the training script will directly fetch the pre-computed, lower-dimensional HNNE features.
+
+```bash
+# COCOSTUFF
+for dimred in "PCA" "RP":
```
Comment (MaxSchambach) [inline README.md:118]: Quick tip for next time: in bash, everything is a string, so there is no need to include the `"` (unless there are some escape characters involved...). Also, you can then access the variable in a  string context as 
```bash
${dimred}_coco_dim${dim}_date${DATE}
```
so you don't need the `"` here either which is more bashonic 🙃

File: README.md (hunk 1 above)
Comment (MaxSchambach) [reply README.md:83]: ah actually, if you do it line-by-line there is no need to include the `;`either, but the `:` need to go  :)

**Review by MaxSchambach — APPROVED** [review]
Thank you, this looks great! Also, lovely to see some light bash scripting going on :) I've left a few minor remarks, otherwise LGTM!

**Comment by axkoenig** [comment]
Many thanks for your comments, I integrated them - the bash scripting looks much cleaner now, cheers!

**Pull request merged**